tests/test-lib/run_all_tests

```diff
@@ -20,17 +20,29 @@ TEST_SET_POS="foo bar"
 TEST_SET_NEG="foo_neg bar"
 UNSTABLE_TESTS=""
 IGNORE_UNSTABLE_TESTS=0
+ret_val=0
 
-# positive test
+echo "running positive TC that should pass"
 CID_FILE_DIR=$(mktemp -d)
 TEST_SUMMARY=""
 TEST_SET=${TEST_SET_POS} ct_run_tests_from_testset "should_pass" >> /dev/null
-(ct_cleanup >> /dev/null)
-test $? -eq 0
+(ct_cleanup >> /dev/null 2>&1)
```

Comment:
Why in brackets? If it fails, then we have a problem somewhere else. We do not need to run it in different shell.

Reply:
This can be moved after condition `if`

Reply:
The subshell is there because the correct beviour of `ct_cleanup` is to exit a shell at its end. That is not usable in this test as we need to collect the result afterwards.
Thus the subshell is used. If the shell in subshell exits with 1 also the return from subshell is 1.

> This can be moved after condition if
What is the reason for moving this? What benefits would it bring? The `ct_run_tests_from_testset` does not return value based on test results, only the TESTSUITE_RESULT is set. Do you mean we should check for TESTSUITE_RESULT here?

Reply:
@phracek?